src/lib.rs

```diff
@@ -733,9 +734,68 @@ fn push_opcode_size(script_size: usize) -> usize {
 
 /// Helper function used by tests
 #[cfg(test)]
-fn hex_script(s: &str) -> bitcoin::Script {
+fn hex_script(s: &str) -> bitcoin::ScriptBuf {
     let v: Vec<u8> = bitcoin::hashes::hex::FromHex::from_hex(s).unwrap();
-    bitcoin::Script::from(v)
+    bitcoin::ScriptBuf::from(v)
+}
+
+/// An absolute locktime that implements `Ord`.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
```

Comment:
You might want to use
```
#[allow(clippy::derive_ord_xor_partial_ord)]
```
Perhaps with a comment (currently missing from `rust-bitcoin`) that its ok to derive eq and implement ord because the enum variants hold the same u32 as returned by `to_u32()` so the impls match as required.

Reply:
Have we added clippy in CI in rust-bitcoin? Perhaps, that gives a green light to add it here too